eZ/Bundle/EzPublishCoreBundle/ApiLoader/RepositoryFactory.php

```diff
@@ -105,6 +105,10 @@ public function buildRepository(PersistenceHandler $persistenceHandler, SearchHa
             new UserReference($this->configResolver->getParameter('anonymous_user_id'))
         );
 
+        $repository->setBackgroundIndexer(
```

Comment:
Is there a reason it cannot be injected into constructor? We're relying on a fact that `$repositoryClass` will have this method, so we can also rely on this class having another optional argument.

Reply:
true, I'll double check why I went that route